onnx/tools/update_model_dims.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from six import string_types
+from typing import Any, List, Text
+from onnx import ModelProto, ValueInfoProto
+
+import onnx.checker
+
+
+def update_inputs_outputs_dims(model, input_dims, output_dims):  # type: (ModelProto, List[List[Any]], List[List[Any]]) -> ModelProto
+    """
+        This function updates the dimension sizes of the model's inputs and outputs to the values
+        provided in input_dims and output_dims. if the dim value provided is negative, a unique dim_param
+        will be set for that dimension.
+
+        Example. if we have the following shape for inputs and outputs:
+                shape(input_1) = ('b', 3, 'w', 'h')
+                and shape(input_2) = ('b', 4)
+                and shape(output)  = ('b', 'd', 5)
+
+            The parameters can be provided as:
+                input_dims = [
+                    ['b', 3, 'w', 'h'],
+                    ['b', 4]
+                ]
+                output_dims = [
+                    ['b', -1, 5]
+                ]
+
+            Putting it together:
+                model = onnx.load('model.onnx')
+                updated_model = update_inputs_outputs_dims(model, input_dims, output_dims)
+                onnx.save(updated_model, 'model.onnx')
+    """
+    def update_dim(tensor, dim, i, j, dim_param_prefix):  # type: (ValueInfoProto, Any, int, int, Text) -> None
+        dim_proto = tensor.type.tensor_type.shape.dim[j]
```

Comment:
Would you check if the existing shape and the provided shape are compatible? Changing `[-1, 3]` to `[2, 3]` is reasonable but I am not sure if you want to allow `[1, 2, 3]` to `[4, 5, 6]`.

Reply:
good catch, added.